src/python/pants/backend/codegen/protobuf/python/rules.py

```diff
@@ -117,6 +119,31 @@ async def generate_python_from_protobuf(
             ),
         )
 
+        if request.protocol_target.get(ProtobufGrpcToggle).value:
+            mypy_info = await Get(PexResolveInfo, VenvPex, mypy_pex)
+
+            # In order to generate stubs for gRPC code, we need mypy-protobuf 2.0 or above.
+            if any(
+                dist_info.project_name == "mypy-protobuf" and dist_info.version.major >= 2
+                for dist_info in mypy_info
+            ):
+                # TODO: merge PEXes together using `pex_path` instead once possible through
+                #  the `VenvPex` abstraction.
+                mypy_pex = await Get(
+                    VenvPex,
+                    VenvPexRequest(
+                        bin_names=[protoc_gen_mypy_script, protoc_gen_mypy_grpc_script],
+                        pex_request=PexRequest(
+                            output_filename="mypy_protobuf.pex",
+                            internal_only=True,
+                            requirements=PexRequirements(
+                                [python_protobuf_subsystem.mypy_plugin_version]
+                            ),
+                            interpreter_constraints=PexInterpreterConstraints(["CPython>=3.5"]),
+                        ),
+                    ),
```

Comment:
It looks like this part is duplicated with the above `PexRequest` and can be extracted into a common variable. But given that I'll be changing this code very soon to use `--pex-path`, it's not a big deal and no need to fix here.